src/objects/grid_obj.f90

```diff
@@ -136,31 +137,31 @@ function my_start(n_global, me, nimg) result(memory_start)
         memory_start = (me-1)*(base_n) + min(me-1,mod(n_global,nimg)) + 1
 
     end function my_start
-
+    
     !> -------------------------------
     !! Generate the domain decomposition mapping and compute the indicies for local memory
     !!
     !! -------------------------------
-    module subroutine set_grid_dimensions(this, nx, ny, nz, nx_extra, ny_extra, halo_width, for_image)
+    module subroutine set_grid_dimensions(this, nx, ny, nz, adv_order, nx_extra, ny_extra, for_image)
       class(grid_t),   intent(inout) :: this
       integer,         intent(in)    :: nx, ny, nz
-      integer,         intent(in), optional :: nx_extra, ny_extra, halo_width, for_image
+      integer,         intent(in), optional :: adv_order, nx_extra, ny_extra, for_image
 
       integer :: nx_e, ny_e, halo_size
       integer :: image
 
       halo_size = kDEFAULT_HALO_SIZE
-      if (present(halo_width)) halo_size = halo_width
+      if (present(adv_order)) halo_size = ceiling(adv_order/2.0)
 
-      image = this_image()
+      image = max(1,FINDLOC(DOM_IMG_INDX,this_image(),dim=1))
```

Comment:
would it make sense for `max(1,FINDLOC(DOM_IMG_INDX,this_image(),dim=1))` to be a function so that this logic is only in one location, not repeated everywhere it is needed?  If it wouldn't change, we could even make `image` (or `my_image_number` or something like that) just be a value in icar_constants that is computed at initialization?

Reply:
Nice catch! This is actually a part of the async I/O changes, but this change slipped in. I'll comment it out for how. But yes, something similar to what you suggested is how it is implemented for async I/O

Reply:
This shouldn't be in the pull request... my current version of the code on the incremental branch doesn't show this change :/ It relates to the async I/O

Reply:
Any thoughts on how/why this is showing up?